lib/cache/add-remote-git.js

```diff
@@ -173,97 +176,124 @@ function fetchRemote (p, u, co, origUrl, cb) {
   )
 }
 
-function resolveHead (p, u, co, origUrl, cb) {
+function resolveHead (params, cb) {
+  log.verbose('resolveHead', 'original treeish', params.treeish)
   git.whichAndExec(
-    [ "rev-list", "-n1", co ],
-    { cwd : p, env : gitEnv() },
+    ['rev-list', '-n1', params.treeish],
+    { cwd: params.checkoutPath, env: gitEnv() },
     function (er, stdout, stderr) {
-      stdout = (stdout + "\n" + stderr).trim()
+      params.exactTreeish = (stdout + '\n' + stderr).trim()
```

Comment:
It does, but I wonder how much that is a real issue, and how much is an urge towards ideological purity, given that this particular `params` never escapes the context of this file. I have an alternative approach that I'm going to try, though, which is pulling all the non-utility parts of the caching flow back into the main function, and setting the individual variables within the scope of the closure. That's how it was originally written, and now that the control flow has been cleaned up a lot, it makes sense to bring it back to that style again.